### PR TITLE
[VarExporter] Add support of PHP enumerations

### LINKED ISSUE #40237: [VarExporter] add support for Enums on PHP 8.1
https://wiki.php.net/rfc/enumerations

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #40237
| License       | MIT
| Doc PR        | _none_

### CODE REVIEW

File: src/Symfony/Component/VarExporter/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+5.3.0
```
Comment (OskarStark) [inline src/Symfony/Component/VarExporter/CHANGELOG.md]:
```suggestion
5.3
```

File: src/Symfony/Component/VarExporter/Tests/VarExporterTest.php [hunk 2]
```diff
@@ -209,6 +210,10 @@ public function provideExport()
         yield ['private-constructor', PrivateConstructor::create('bar')];
 
         yield ['php74-serializable', new Php74Serializable()];
+
+        if (\PHP_VERSION_ID >= 80100) {
+            yield ['unit-enum', FooUnitEnum::Bar, true];
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarExporter/Tests/VarExporterTest.php]: Let's try with `[FooUnitEnum::Bar]` instead, this should spot that there is a missing check in `Exporter` on the line that looks like:
`} elseif (!\is_object($value) && !$value instanceof \__PHP_Incomplete_Class) {`

**Review by nicolas-grekas — COMMENTED** [review]
I didn't expect this patch to be that simple, thanks for opening :)
Can you please rebase for 4.4? We consider support for newer PHP versions as a bug fix.

File: src/Symfony/Component/VarExporter/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+5.3
+-----
```
Comment (OskarStark) [inline src/Symfony/Component/VarExporter/CHANGELOG.md]:
```suggestion
---
```

File: src/Symfony/Component/VarExporter/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,11 @@
...
+
+ * Add support of PHP 8.1 enumerations
```
Comment (OskarStark) [inline src/Symfony/Component/VarExporter/CHANGELOG.md]:
```suggestion
 * Add support of PHP enumerations
```

File: src/Symfony/Component/VarExporter/Tests/VarExporterTest.php (hunk 2 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/VarExporter/Tests/VarExporterTest.php]: Done!

File: src/Symfony/Component/VarExporter/CHANGELOG.md (hunk 3 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/VarExporter/CHANGELOG.md]: Updated after changing base to 4.4 :+1:

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @alexandre-daubois.

**Pull request merged**